crates/pathfinder/src/rpc/v02/method/get_transaction_by_hash.rs

```diff
@@ -145,10 +145,7 @@ mod tests {
 
     #[tokio::test]
     async fn pending() {
-        let context = RpcContext::for_tests();
-        let pending_data = crate::rpc::tests::create_pending_data(context.storage.clone()).await;
-        let context = context.with_pending_data(pending_data);
-        let context = std::sync::Arc::new(context);
+        let context = std::sync::Arc::new(RpcContext::for_tests_with_pending().await);
```

Comment:
Considering these will always be `Arc::new` should we fold that in as well?

Reply:
And I guess similarly for `RpcContext::for_tests` as well?

Reply:
That makes composing these harder -- but now I've realized that we don't use other `with_*` methods in tests, so there's no need for that. I'll add `Arc`.